tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -80,6 +84,27 @@ public bool TryExtractGitMetadata([NotNullWhen(true)] out GitMetadata? gitMetada
         }
     }
 
+    private void PropagateGitMetadataToTheProfiler(GitMetadata gitMetadata)
+    {
+        try
+        {
+            // Avoid P/Invoke if the profiler is not ready (for obvious reason)
+            // but also if both repository url and commit sha are empty
+            if (Profiler.Instance.Status.IsProfilerReady &&
+                (!string.IsNullOrWhiteSpace(gitMetadata.RepositoryUrl) || !string.IsNullOrWhiteSpace(gitMetadata.CommitSha)))
+            {
+                NativeInterop.SetGitMetadata(RuntimeId.Get(), gitMetadata.RepositoryUrl, gitMetadata.CommitSha);
+            }
+        }
+        catch (Exception ex)
+        {
+            // We failed to retrieve the runtime from native this can be because:
+            // - P/Invoke issue (unknown dll, unknown entrypoint...)
+            // - We are running in a partial trust environment
+            Log.Warning(ex, "Failed to set git metadata for native.");
```

Comment:
Maybe a different message could be more explicit for support teams.
```suggestion
            Log.Warning(ex, "Failed to share git metadata with Continuous Profiler.");
```

Reply:
good point